cfme/tests/infrastructure/test_vm_power_control.py

```diff
@@ -408,11 +408,31 @@ def test_suspend(self, verify_vm_running, vm):
         vm.action.suspend()
         wait_for(lambda: vm.power_state == "suspended", num_sec=300, delay=5, fail_func=vm.reload)
 
+    def test_reset(self, verify_vm_running, vm):
+        assert "reset" in vm.action
+        vm.action.reset()
+        wait_for(lambda: vm.power_state == "on", num_sec=300, delay=5, fail_func=vm.reload)
+
+    def test_restart(self, verify_vm_running, vm):
+        assert "restart" in vm.action
+        vm.action.restart()
+        wait_for(lambda: vm.power_state == "on", num_sec=300, delay=5, fail_func=vm.reload)
```

Comment:
What I was thinking is - if you do a restart and check if the VM is on, it might **still** be on. So I would first expect a wait_for for the VM to turn off and then a wait_for for it to get back up to a running state.